Revit_Engine/Compute/RevitDiffing.cs

```diff
@@ -155,24 +155,59 @@ private static Diff Diffing(IEnumerable<object> pastObjects, IEnumerable<object>
             DiffingConfig diffConfigClone = GetRevitDiffingConfig(rcc);
 
             // Check if input objects all have RevitIdentifiers assigned.
-            IEnumerable<IBHoMObject> revitBHoMObjects_past = pastObjects.OfType<IBHoMObject>().Where(obj => obj.GetRevitIdentifiers() != null);
-            IEnumerable<IBHoMObject> revitBHoMObjects_following = followingObjects.OfType<IBHoMObject>().Where(obj => obj.GetRevitIdentifiers() != null);
+            IEnumerable<RevitIdentifiers> pastIdFragments = pastObjects.OfType<IBHoMObject>().Select(obj => obj.GetRevitIdentifiers()).Where(x => x != null);
```

Comment:
Do you need to change this section of the method? It works, I would rather not touch it if possible, mainly to avoid the risk of breaking something

Reply:
technically do not _need_ to change it, but I do need the line I added to be able to get the ids out. The previous collections were only used for a count check, and I cannot see any scenario where the count of the new collections would be different than the new ones. The only difference is the return type, which is used now, but was not before.

Happy to chat through if you think there are cases where the count could be different, but I honestly cant see it being possible, even if I tried to force it.

Reply:
tl;dr: I think this is actually fine.

Longer reasoning: I think that I originally configured the method like this (getting the revit objects having a revit identifier, rather than directly getting the revit identifiers) because I thought that one could fall back on other Diffing methods not requiring a fragment Id attached, for the objects not owning one. However, as the solution took form, I formalized the `RevitDiffing()` method as only working with Fragments anyway, leaving the responsibility of non-fragment based diffing methodologies to the base Diffing_Engine.